### PR TITLE
Add load_extension_from_path function

### PR DESCRIPTION
**Pull request by Sodynoizz**
## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

**Comment by Sodynoizz** [comment]
This function will help to automatically load all of the extensions in that path (.py files) instead of using for loop manually.
I hope that **load_extension_from_path** will get a good review from staff.

**Comment by baronkobama** [comment]
I believe this PR will implement the feature requested in #946, and therefore close that issue. If possible, could we meet something more along with the ideal solution in that feature request, wherein, this directory loading is built-in to the standard bot.load_extension? Also, currently you can only load with `dir/dir`-esque syntax and not the standard `dir.dir` syntax.

**Comment by Sodynoizz** [comment]
> I believe this PR will implement the feature requested in #946, and therefore close that issue. If possible, could we meet something more along with the ideal solution in that feature request, wherein, this directory loading is built-in to the standard bot.load_extension? Also, currently you can only load with `dir/dir`-esque syntax and not the standard `dir.dir` syntax.

> I think this is a great idea to built-in to load_extension but it depends on staff to approve this idea.

**Comment by Lulalaby** [comment]
> I think this is a great idea to built-in to load_extension but it depends on staff to approve this idea.

*approve*

File: discord/cog.py [hunk 1]
```diff
@@ -740,6 +742,34 @@ def _resolve_name(self, name: str, package: Optional[str]) -> str:
         except ImportError as e:
             raise errors.ExtensionNotFound(name) from e
 
+    def load_extension_from_path(
```
Comment (plun1331) [inline discord/cog.py:745]: This could be `load_extensions_from_path` to reflect loading multiple extensions

File: discord/colour.py [hunk 2]
```diff
@@ -329,7 +329,15 @@ def nitro_pink(cls: Type[CT]) -> CT:
         .. versionadded:: 2.0
         """
         return cls(0xF47FFF)
-
+    
+    @classmethod
```
Comment (plun1331) [inline discord/colour.py:333]: Unrelated to this PR, should be reverted and put in a different PR if you want it to be added.

File: discord/colour.py [hunk 3]
```diff
@@ -329,7 +329,6 @@ def nitro_pink(cls: Type[CT]) -> CT:
         .. versionadded:: 2.0
         """
         return cls(0xF47FFF)
```
Comment (Lulalaby) [inline discord/colour.py:331]:
```suggestion
        return cls(0xF47FFF)
        
```

File: discord/colour.py [hunk 4]
```diff
@@ -329,7 +329,7 @@ def nitro_pink(cls: Type[CT]) -> CT:
...
-
+
```
Comment (Lulalaby) [inline discord/colour.py:332]:
```suggestion

```

File: discord/cog.py (hunk 1 above)
Comment (Sodynoizz) [reply discord/cog.py:744]: Okay i will fix it

Comment (Sodynoizz) [reply discord/cog.py]: Fixed

File: discord/cog.py [hunk 5]
```diff
@@ -23,7 +23,8 @@
 DEALINGS IN THE SOFTWARE.
 """
 from __future__ import annotations
-
```
Comment (baronkobama) [inline discord/cog.py:26]: This space (between `from __future__ import annotations` and `import importlib`/`import os`) should still be here.

File: discord/cog.py [hunk 6]
```diff
@@ -23,7 +23,8 @@
...
+import os
+import pathlib
 import importlib
 import inspect
 import sys
```
Comment (baronkobama) [inline discord/cog.py:30]: Alphabetize the newly imported modules.
```suggestion
import importlib
import inspect
import os
import pathlib
import sys
```

File: discord/colour.py [hunk 7]
```diff
@@ -329,7 +329,6 @@ def nitro_pink(cls: Type[CT]) -> CT:
         .. versionadded:: 2.0
         """
         return cls(0xF47FFF)
-
```
Comment (baronkobama) [inline discord/colour.py:332]: This space (between `return cls(0xF47FFF)` and `@classmethod`) should still be here.

**Review by baronkobama — COMMENTED** [review]
Other than comments, and assuming we don't build directory loading into load_extension, looks good.

**Comment by Lulalaby** [comment]
> > I think this is a great idea to built-in to load_extension but it depends on staff to approve this idea.
> 
> 
> 
> *approve*

Well as said, we approve this. So please

File: discord/cog.py [hunk 8]
```diff
@@ -739,16 +742,41 @@ def _resolve_name(self, name: str, package: Optional[str]) -> str:
...
+
+        path = paths[0]
+
+        if isinstance(path, str):
+            path = pathlib.Path(path)
+
+        try:
+            path = path.resolve().relative_to(pathlib.Path.cwd())
+        except ValueError:
+            raise ValueError(f"'{path}' must be relative to the working directory") from None
```
Comment (EmreTech) [inline discord/cog.py:761]: `pathlib.Path` should have a function that allows for you to check if the path provided is absolute. Use that instead of this check.

File: discord/cog.py [hunk 9]
```diff
@@ -739,16 +742,41 @@ def _resolve_name(self, name: str, package: Optional[str]) -> str:
...
+            if must_exist:
+                raise FileNotFoundError(f"'{path}' is not an existing directory")
+            return
 
+        glob = path.rglob if recursive else path.glob
+        for ext_path in glob("[!_]*.py"):
+            ext = str(ext_path.with_suffix("")).replace(os.sep, ".")
+            self.load_extension(ext)
+        
+    def load_extension(self, name: t.Union[str , pathlib.Path], *, package: Optional[str] = None) -> None:
```
Comment (EmreTech) [inline discord/cog.py:773]: Why is this `t.Union` and not just `Union`

File: discord/cog.py [hunk 10]
```diff
@@ -739,16 +742,41 @@ def _resolve_name(self, name: str, package: Optional[str]) -> str:
         except ImportError:
             raise errors.ExtensionNotFound(name)
 
-    def load_extension(self, name: str, *, package: Optional[str] = None) -> None:
-        """Loads an extension.
+    def load_extensions_from_path(
+        self, *paths: Union[str, pathlib.Path], recursive: bool = False, must_exist: bool = True
+    ) -> None:
```
Comment (EmreTech) [inline discord/cog.py:747]: I feel like this should return a list of all of the extensions loaded just in case the bot developer wants to unload or reload all of them.

**Review by EmreTech — COMMENTED** [review]
Looks good, here are some suggestions
